frontend/src/metabase/lib/browser.ts

```diff
@@ -2,16 +2,20 @@ import querystring from "querystring";
 
 import { safeJsonParse } from "metabase/lib/utils";
 
-function parseQueryStringOptions(s) {
-  const options = querystring.parse(s);
+function parseQueryStringOptions(s: string) {
+  const options: Record<string, string | string[] | boolean | undefined> =
```

Comment:
I don't really mind but there is also the `Query` object in `history` which has this type - I use it mainly to reduce the number of times we have to write this obtuse type

Reply:
The return value of parse is actually typed something like that already, but we need to add `boolean` because we do safeJsonParse a few lines below :(